lib/Demangling/OldRemangler.cpp

```diff
@@ -1215,14 +1216,18 @@ void Remangler::mangleProtocolList(Node *node) {
   mangleProtocolListWithoutPrefix(node);
 }
 
-void Remangler::mangleProtocolListWithoutPrefix(Node *node) {
+void Remangler::mangleProtocolListWithoutPrefix(Node *node,
+                                                Node *additionalProto) {
   assert(node->getKind() == Node::Kind::ProtocolList);
   assert(node->getNumChildren() == 1);
   auto typeList = node->begin()[0];
   assert(typeList->getKind() == Node::Kind::TypeList);
   for (auto &child : *typeList) {
     mangleProtocolWithoutPrefix(child);
   }
+  if (additionalProto) {
+    mangleProtocolWithoutPrefix(additionalProto);
```

Comment:
There's one more nitpick. If you have `P & AnyObject` for some protocol P, then in Swift 3, it will mangle as either `P & AnyObject` or `AnyObject & P`, depending on how ProtocolType::canonicalizeProtocols() ordered AnyObject vs P. :-)